lib/segment/src/index/field_index/bool_index/mmap_bool_index.rs

```diff
@@ -79,21 +81,32 @@ impl MmapBoolIndex {
             indexed_count: 0,
             trues_count: 0,
             falses_count: 0,
+            populated: populate,
         })
     }
 
+    fn make_conditioned_counter<'a>(
+        &self,
+        hw_counter: &'a HardwareCounterCell,
+    ) -> ConditionedCounter<'a> {
+        let on_disk = !self.populated; // Measure if on disk.
+        ConditionedCounter::new(on_disk, hw_counter)
+    }
```

Comment:
Can't we return a dummy counter here somehow, without a boolean condition inside the counter itself?

Reply:
We probably can but we'll use this in more places so I thought having a separate logic with known semantics makes the code more readable and easier to implement.

Reply:
Also there are places where we would run into lifetime issues with local disposable hardware counters.
An example for this is the `ChunkReader` in #6309